### PR TITLE
fix(modal): backdrop click focus to title

### PR DESCRIPTION
**Pull request by valentin-mladenov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
Clicking on modal/wizard backdrop alows the focus trap to be escaped.

Issue Number: CDE-910

## What is the new behavior?
Clicking on modal/wizard backdrop returns focus to the modal title keeping the focus trap.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/angular/src/modal/modal.spec.ts [hunk 1]
```diff
@@ -258,6 +259,27 @@ describe('Modal', () => {
...
+    const titleWrapperElement: HTMLElement = compiled.querySelector('div.modal-title-wrapper');
+    const focusStealButton: HTMLElement = compiled.querySelector('button.btn.to-focus');
+
+    fixture.componentInstance.staticBackdrop = true;
+    fixture.detectChanges();
+
+    // Just make sure we have the "x" to close the modal,
+    // because this is different from the clrModalClosable option.
+    expect(compiled.querySelector('.close')).not.toBeNull();
+    expect(titleWrapperElement).toEqual(document.activeElement as HTMLElement);
```
Comment (kevinbuhmann) [inline projects/angular/src/modal/modal.spec.ts:273]:
```suggestion
    expect(document.activeElement ).toEqual(titleWrapperElement);
```

Assertions should be written as `expect(actualValue).toBe(expected value);`.

File: projects/angular/src/modal/modal.spec.ts [hunk 2]
```diff
@@ -258,6 +259,27 @@ describe('Modal', () => {
...
+
+    focusStealButton.focus();
+    expect(focusStealButton).toEqual(document.activeElement as HTMLElement);
```
Comment (kevinbuhmann) [inline projects/angular/src/modal/modal.spec.ts:276]:
```suggestion
    expect(document.activeElement ).toEqual(focusStealButton);
```

Assertions should be written as `expect(actualValue).toBe(expected value);`.

File: projects/angular/src/modal/modal.spec.ts [hunk 3]
```diff
@@ -258,6 +259,27 @@ describe('Modal', () => {
...
+
+    backdrop.click();
+    flushAndExpectOpen(fixture, true);
+    expect(titleWrapperElement).toEqual(document.activeElement as HTMLElement);
```
Comment (kevinbuhmann) [inline projects/angular/src/modal/modal.spec.ts:280]:
```suggestion
    expect(document.activeElement ).toEqual(titleWrapperElement);
```

Assertions should be written as `expect(actualValue).toBe(expected value);`.

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]

File: projects/angular/src/modal/modal.spec.ts (hunk 3 above)
Comment (valentin-mladenov) [reply projects/angular/src/modal/modal.spec.ts]: corrected in commit 31f7c5e

File: projects/angular/src/modal/modal.spec.ts (hunk 2 above)
Comment (valentin-mladenov) [reply projects/angular/src/modal/modal.spec.ts]: corrected in commit 31f7c5e

File: projects/angular/src/modal/modal.spec.ts (hunk 1 above)
Comment (valentin-mladenov) [reply projects/angular/src/modal/modal.spec.ts]: corrected in commit 31f7c5e

File: projects/angular/src/modal/modal.spec.ts [hunk 4]
```diff
@@ -258,6 +259,27 @@ describe('Modal', () => {
...
+    const titleWrapperElement: HTMLElement = compiled.querySelector('div.modal-title-wrapper');
+    const focusStealButton: HTMLElement = compiled.querySelector('button.btn.to-focus');
+
+    fixture.componentInstance.staticBackdrop = true;
+    fixture.detectChanges();
+
+    // Just make sure we have the "x" to close the modal,
+    // because this is different from the clrModalClosable option.
+    expect(compiled.querySelector('.close')).not.toBeNull();
+    expect(document.activeElement).toEqual(titleWrapperElement);
```
Comment (kevinbuhmann) [inline projects/angular/src/modal/modal.spec.ts:273]:
```suggestion
    expect(document.activeElement).toBe(titleWrapperElement);
```

I think these should be `toB` (strict equality).

**Review by kevinbuhmann — APPROVED** [review]
This is fine, but maybe change `toEqual` to `toBe`.

**Comment by valentin-mladenov** [comment]
> This is fine, but maybe change `toEqual` to `toBe`.

Corrected in commit 4f23904

**Review by kevinbuhmann — APPROVED** [review]

**Pull request merged**